main/src/com/google/refine/expr/functions/ToDate.java

```diff
@@ -168,8 +167,7 @@ private OffsetDateTime parse(String o1, Locale locale, List<String> formats) {
         } else {
             try {
                 return javax.xml.bind.DatatypeConverter.parseDateTime(o1).getTime().toInstant()
-                		.plusSeconds(ZonedDateTime.now().getOffset().getTotalSeconds())
-                		.atOffset(ZoneOffset.of("Z"));
+                        .truncatedTo(ChronoUnit.DAYS).atOffset(ZoneOffset.of("Z"));
```

Comment:
It is not clear to me that this new version is equivalent, for instance of datetime objects with precision down to the second. You are correct that the motivation for this `.plusSeconds()` was to preserve the date part across timezones but I think this code also handles timestamps with a higher precision than day.

Reply:
Can you add a (failing) test that demonstrates the problem? I'm not sure I follow what you're trying to say.

My understanding of the intent was to make sure that `toDate("1984")` ends up pinned to midnight, rather than offset by the number of hours that local time is from UTC. The old code apparently doesn't account for summer time / Daylight Savings Time so, on my system, it ends up at 1 hour past midnight rather than at midnight. The new code fixes that. I'm open to other solutions, but I'm trying to resist having to learn all of the complexities of the new Java date/time system.

Reply:
I've got a better understanding of what the original problem was and have a better fix. The issue occurred when the test date to be parsed was on one side of the Daylight Savings Time boundary and the current date was on the opposite side, resulting in the offset computed using `now()` to be incorrect. I switched the offset calculation to use the parsed date.

Somehow, in the mean time, the default timezone for Java switched itself from "EDT" to "UTC" so I also changed the test to use a fixed, non-UTC timezone to hopefully catch any other UTC dependencies in the future (although it won't catch temporal shifts like DST).